src/components/AutoExpandingTextArea.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { FC, useLayoutEffect } from "react";
+import { Textarea, TextareaProps } from "@canonical/react-components";
+
+type Props = {
+  dynamic?: boolean;
+  value?: string;
+} & TextareaProps;
+
+const AutoExpandingTextArea: FC<Props> = (props: Props) => {
+  const { dynamic, id, value, rows } = props;
+
+  useLayoutEffect(() => {
+    if (dynamic) {
+      const textArea = document.getElementById(id || "");
+      if (textArea) {
+        textArea.style.height = "0px";
+        const scrollHeight = textArea.scrollHeight;
+        textArea.style.height = `${scrollHeight + 1}px`;
+      }
+    }
+  }, [value]);
+
+  return (
+    <div>
+      <Textarea
+        {...(props as TextareaProps)}
+        rows={dynamic ? undefined : rows}
```

Comment:
If a user opt to not use the dynamicHeight functionality of this component, then rows should still be pass in as a prop. Although I don't think there is a need to override rows here since I'm already destructuring props above.